node/src/components/rpc_server/http_server.rs

```diff
@@ -90,27 +90,35 @@ pub(super) async fn run<REv: ReactorEventT>(
         Ok::<_, Rejection>(error_response)
     });
 
+    let service_routes = rpc_put_deploy
+        .or(rpc_get_block)
+        .or(rpc_get_block_transfers)
+        .or(rpc_get_state_root_hash)
+        .or(rpc_get_item)
+        .or(rpc_get_balance)
+        .or(rpc_get_deploy)
+        .or(rpc_get_peers)
+        .or(rpc_get_status)
+        .or(rpc_get_era_info)
+        .or(rpc_get_auction_info)
+        .or(rpc_get_account_info)
+        .or(rpc_get_rpcs)
+        .or(rpc_get_dictionary_item)
+        .or(rpc_query_global_state)
+        .or(unknown_method)
+        .or(parse_failure);
+
+    // Supports content negotiation for gzip responses. This is an interim fix until
+    // https://github.com/seanmonstar/warp/pull/513 moves forward.
+    let service_routes_gzip = warp::header::exact("accept-encoding", "gzip")
+        .and(service_routes.clone())
+        .with(warp::compression::gzip());
```

Comment:
Is this adding the `Content-Encoding` header automatically?

Reply:
Yes, `warp`'s `compression` adds this to the response. Warp doesn't yet support automatically inspecting the `accept-encoding` header, so this does the check manually.